client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/internal/resource/ResourceClientTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.internal.resource;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import io.micronaut.runtime.server.EmbeddedServer;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Note that this class must be public for the subscriptions on the event bus to
+ * work. SonarQube complains about this though.
+ *
+ * @since 0.5.0
+ */
+@MicronautTest
+public class ResourceClientTest {
+
+    private static final String TOKEN = "abcd-1";
+
+    @Inject
+    EmbeddedServer embeddedServer;
+
+    private ObjectMapper objectMapper;
+    private WebSocketClient resourceClient;
+
+    private int port;
+
+    @BeforeEach
+    void setup() {
+        this.port = embeddedServer.getPort();
+        this.objectMapper = new ObjectMapper().registerModule(new Jdk8Module());
+
+        this.resourceClient = WebSocketClient
+            .createResourceClient(b -> b
+                .httpClient(HttpClient.newHttpClient())
+                .token(TOKEN)
+                .uri(URI.create("ws://localhost:" + port + "/cluster/filteredResource/name/%25t"))
```

Comment:
```suggestion
                .uri(URI.create("ws://localhost:" + port + "/cluster/filteredResource/resource/%25t"))
```
A lot of corrections to do for these paths to stay consistent with actual service implementation

Reply:
This is now `"/cluster/resource/%25t"`. But these urls only point to the internal server for testing, so they could be different.